### PR TITLE
Refactor base path and path BBE

### PR DESCRIPTION
**Pull request by chamil321**

### CODE REVIEW

File: examples/http-absolute-path-and-path/absolute_path_and_path.bal [hunk 1]
```diff
@@ -1,8 +1,15 @@
 import ballerina/http;
 import ballerina/log;
 
-service /foo on new http:Listener(9090) {
+// The `absolute-resource-path` identifier represent the absolute path to the service.  When bound to a listener
```
Comment (praneesha) [inline examples/http-absolute-path-and-path/absolute_path_and_path.bal:4]:
```suggestion
// The `absolute-resource-path` identifier represents the absolute path to the service.  When bound to a listener
```

File: examples/http-absolute-path-and-path/absolute_path_and_path.bal [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+// endpoint, the service will be accessible at the specified path. If the path is omitted, then it defaults to `/`.
+// A string literal also can represent the absolute path. Eg: ``"/foo"`
```
Comment (praneesha) [inline examples/http-absolute-path-and-path/absolute_path_and_path.bal:6]:
```suggestion
// A string literal also can represent the absolute path. E.g., ``"/foo"`.
```

File: examples/http-absolute-path-and-path/absolute_path_and_path.bal [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+// The `type descriptor` represents the respective type of the service. Eg. `http:Service`
```
Comment (praneesha) [inline examples/http-absolute-path-and-path/absolute_path_and_path.bal:7]:
```suggestion
// The `type descriptor` represents the respective type of the service. E.g., `http:Service`.
```

File: examples/http-absolute-path-and-path/absolute_path_and_path.bal [hunk 4]
```diff
@@ -1,8 +1,15 @@
...
+service http:Service /foo on new http:Listener(9090) {
 
+    // The `accessor-name` identifier (`post`) confines the resource to the specified HTTP methods. In this instance,
+    // only `POST` requests are allowed. The `default` accessor name can be used to match with any HTTP method.
+    // The `resource-name` identifier associates the relative path to the service object's path. Eg. `bar`
```
Comment (praneesha) [inline examples/http-absolute-path-and-path/absolute_path_and_path.bal:12]:
```suggestion
    // The `resource-name` identifier associates the relative path to the service object's path. E.g., `bar`.
```

File: examples/http-absolute-path-and-path/absolute_path_and_path.description [hunk 5]
```diff
@@ -1,5 +1,5 @@
 // Ballerina supports writing RESTful services according to the JAX-RS specification. 
-// You can use `BasePath`, `Path`, and HTTP verb annotations such as `POST` and `GET` to
-// constrain your service in a RESTful manner.<br/><br/>
+// You can use `absolute-resource-path` and `resource-name` to access a resource function while the `accessor-name`
```
Comment (praneesha) [inline examples/http-absolute-path-and-path/absolute_path_and_path.description:2]:
```suggestion
// You can use the `absolute-resource-path` and `resource-name` to access a resource function while the `accessor-name`,
```

File: examples/http-absolute-path-and-path/absolute_path_and_path.description [hunk 6]
```diff
@@ -1,5 +1,5 @@
...
+// which is HTTP verb as `post` and `get` to constrain your resource function in a RESTful manner.<br/><br/>
```
Comment (praneesha) [inline examples/http-absolute-path-and-path/absolute_path_and_path.description:3]:
```suggestion
// which is an HTTP verb as `post` and `get` to constrain your resource function in a RESTful manner.<br/><br/>
```

File: examples/query-path-matrix-param/query_path_matrix_param.bal [hunk 7]
```diff
@@ -4,6 +4,7 @@ import ballerina/log;
 service /sample on new http:Listener(9090) {
 
     // The `PathParam` and `QueryParam` parameters extract values from the request URI.
+    // Path param is defined as a part of resource path along with the type.
```
Comment (praneesha) [inline examples/query-path-matrix-param/query_path_matrix_param.bal:7]:
```suggestion
    // Path param is defined as a part of the resource path along with the type.
```

File: examples/index.json [hunk 8]
```diff
@@ -832,6 +832,10 @@
                 "name": "Client Data Binding",
                 "url": "http-client-data-binding"
             },
+            {
```
Comment (niveathika) [inline examples/index.json:835]: Shouldn't we remove the older bbe from index.json?

Comment (chamil321) [reply examples/index.json:835]: It's already been removed.

File: examples/http-absolute-path-and-path/absolute_path_and_path.bal [hunk 9]
```diff
@@ -1,8 +1,15 @@
 import ballerina/http;
 import ballerina/log;
 
-service /foo on new http:Listener(9090) {
+// The `absolute-resource-path` identifier represents the absolute path to the service.  When bound to a listener
+// endpoint, the service will be accessible at the specified path. If the path is omitted, then it defaults to `/`.
+// A string literal also can represent the absolute path. E.g., ``"/foo"`.
```
Comment (chamil321) [inline examples/http-absolute-path-and-path/absolute_path_and_path.bal:6]:
```suggestion
// A string literal also can represent the absolute path. E.g., `"/foo"`.
```

File: examples/index.json (hunk 8 above)
Comment (niveathika) [reply examples/index.json:835]: Ack

**Review by niveathika — APPROVED** [review]

**Pull request merged**